primitives/src/lib.rs

```diff
@@ -252,8 +251,9 @@ pub struct InterExecSchedule<Account, Balance> {
     pub phases: Vec<ExecPhase<Account, Balance>>,
 }
 
+/// TODO: need a currency shim for base currency here
 pub trait EscrowTrait: frame_system::Config + pallet_sudo::Config {
-    type Currency: Currency<Self::AccountId>;
+    type Currency: ReservableCurrency<Self::AccountId>;
```

Comment:
hopefully doesnt kill anything, needed due to the coupling everywhere, although my PR does fix some coupling this still needs to be reservable, for now.